torch/csrc/jit/runtime/autodiff.cpp

```diff
@@ -14,13 +14,75 @@
 #include <torch/csrc/jit/runtime/symbolic_script.h>
 #include <algorithm>
 #include <memory>
+#include <mutex>
+#include "ATen/core/interned_strings.h"
+#include "jit/ir/ir.h"
+
 
 namespace torch {
 namespace jit {
 
 using value_map = std::unordered_map<Value*, Value*>;
 using value_set = std::unordered_set<Value*>;
 
+struct InlineExprsInBwdPasses {
+  std::vector<std::function<void(std::shared_ptr<Graph>, Block* )>> passes_;
+  std::mutex mutex_;
+};
+
+static InlineExprsInBwdPasses& getInlineExprsInBwdPasses() {
+  static InlineExprsInBwdPasses inline_passes;
+  return inline_passes;
+}
+
+void registerInlineExprsInBwdPass(std::function<void(std::shared_ptr<Graph>, Block*)> pass) {
+  auto& inline_passes = getInlineExprsInBwdPasses();
+  std::lock_guard<std::mutex> lock(inline_passes.mutex_);
+  inline_passes.passes_.push_back(std::move(pass));
+}
+
+RegisterInlineExprsInBwdPass::RegisterInlineExprsInBwdPass(std::function<void(std::shared_ptr<Graph>, Block*)> pass) {
+  registerInlineExprsInBwdPass(std::move(pass));
+}
+
+
+static void inlineProfileNodes_(Block* block) {
```

Comment:
Very naive question, since profiling is done after autodiff, does it really matter that we insert profiling nodes again?

Or is this prior to guard/mark requires_grad?